notifications/src/main/kotlin/com/d3/notifications/handler/CommandHandler.kt

```diff
@@ -16,35 +19,60 @@ abstract class CommandHandler {
 
     /**
      * Handles given command if it's possible
-     * @param commandWithTx command to handle
+     * @param irohaCommand command to handle
      */
-    fun handleCommand(commandWithTx: CommandWithTx) {
+    fun handleCommand(irohaCommand: IrohaCommand) {
         try {
-            if (ableToHandle(commandWithTx)) {
-                handle(commandWithTx)
+            if (ableToHandle(irohaCommand)) {
+                handle(irohaCommand)
             }
         } catch (e: Exception) {
-            logger.error("Cannot handle command ${commandWithTx.command}", e)
+            logger.error("Cannot handle command ${irohaCommand.command}", e)
         }
     }
 
     /**
      * The main handling logic
-     * @param commandWithTx - command to handle
+     * @param irohaCommand - command to handle
      */
-    protected abstract fun handle(commandWithTx: CommandWithTx)
+    protected abstract fun handle(irohaCommand: IrohaCommand)
 
     /**
      * Handling predicate. Checks it's possible to handle
-     * @param commandWithTx - command to check
+     * @param irohaCommand - command to check
      * @return true if possible to handle, false otherwise
      */
-    protected abstract fun ableToHandle(commandWithTx: CommandWithTx): Boolean
+    protected abstract fun ableToHandle(irohaCommand: IrohaCommand): Boolean
 
     companion object : KLogging()
 }
 
-data class CommandWithTx(val command: Commands.Command, val tx: TransactionOuterClass.Transaction)
+/**
+ * Data class that represents Iroha command
+ * @param command - command itself
+ * @param tx - tx that has given command inside it
+ * @param block - block that has given command and tx inside it
+ */
+data class IrohaCommand(
+    val command: Commands.Command,
+    val tx: TransactionOuterClass.Transaction,
+    val block: BlockOuterClass.Block
+) {
+    /**
+     * Returns index of the current tx in the current block
+     */
+    fun getTxIndex(): Int {
+        var txIndex = 0
+        block.blockV1.payload.transactionsList.forEach {
+            if (Arrays.equals(Utils.hash(tx), Utils.hash(it))) {
```

Comment:
you don't need to compute tx hash each iteration
why do you use `Arrays`
is it legal to use `forEach` instead of classic iteration (maybe in Kotlin yes)

Reply:
Done